homeassistant/components/tautulli/__init__.py

```diff
@@ -1 +1,85 @@
-"""The tautulli component."""
+"""The Tautulli integration."""
+from __future__ import annotations
+
+from pytautulli import PyTautulli, exceptions
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PATH,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONF_MONITORED_USERS, DATA_KEY_COORDINATOR, DEFAULT_NAME, DOMAIN
+from .coordinator import TautulliDataUpdateCoordinator
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Tautulli from a config entry."""
+    api_client = PyTautulli(
+        api_token=entry.data[CONF_API_KEY],
+        hostname=entry.data[CONF_HOST],
+        session=async_get_clientsession(hass, entry.data[CONF_VERIFY_SSL]),
+        verify_ssl=entry.data[CONF_VERIFY_SSL],
+        port=entry.data[CONF_PORT],
+        ssl=entry.data[CONF_SSL],
+        base_api_path=entry.data[CONF_PATH],
+    )
+    try:
+        await api_client.async_get_activity()
+    except exceptions.PyTautulliConnectionException as ex:
+        raise ConfigEntryNotReady(ex) from ex
+    except (exceptions.PyTautulliAuthenticationException) as ex:
+        raise ConfigEntryAuthFailed(ex) from ex
+
+    cordnator = TautulliDataUpdateCoordinator(hass, api_client)
+    await cordnator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {DATA_KEY_COORDINATOR: cordnator}
+    if not entry.options:
+        options = {CONF_MONITORED_USERS: entry.data.get(CONF_MONITORED_USERS)}
+        hass.config_entries.async_update_entry(entry, options=options)
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+    return unload_ok
+
+
+class TautulliEntity(CoordinatorEntity):
+    """Defines a base Tautulli entity."""
+
+    def __init__(
+        self,
+        coordinator: TautulliDataUpdateCoordinator,
+        name: str,
+        server_unique_id: str,
+    ) -> None:
+        """Initialize the Tautulli entity."""
+        super().__init__(coordinator)
+        self._attr_name = name
+        self._server_unique_id = server_unique_id
+        self._attr_device_info = DeviceInfo(
+            configuration_url=coordinator.api_client._host.base_url,
```

Comment:
`_host` is private
The lib should expose this as a `host_configuration` property.

Reply:
I'm having trouble following this suggestion but if we are to use `CONF_URL` we could just use that instead.

Reply:
There needs to be a property like `PyTautulli.host_configuration` that returns `PyTautulli._host` if you want to use this here.

Reply:
I'm assuming this does not currently exist? 🤷‍♂️

Reply:
That would be correct, another option is to construct a `PyTautulliHostConfiguration` during init and pass that to the coordinator or base entity